### PR TITLE
feat: add payment challenge

### PR DESCRIPTION
**Pull request by cbourgois**
### Description of the Change

cb95e18 — feat: add payment challenge

### Related PRs

- [x] https://github.com/ovh-ux/ovh-api-services/pull/177

### CODE REVIEW

File: src/payment-method-legacy.js [hunk 1]
```diff
@@ -261,6 +277,19 @@ export default class OvhPaymentMethodLegacy {
       }).$promise;
   }
 
+  /**
+   *  Challenge the given payment mean
+   *  @param  {Object} paymentMean The paymentMean object to delete
+   *  @param  {Object} challenge The challenge value
+   *  @return {Promise}
+   */
+  challengePaymentMean(paymentMean, challenge) {
```
Comment (antleblanc) [inline src/payment-method-legacy.js]: Here is a destructuring assignment suggestion:

```js
  challengePaymentMean(paymentMean, challenge) {
    const { id, paymentType } = paymentMean;
    return this.getPaymentMeanResource(paymentType)
      .challenge({
        id,
      }, { challenge }).$promise;
  }
```

File: src/payment-method.service.js [hunk 2]
```diff
@@ -170,6 +170,24 @@ export default class OvhPaymentMethodService {
     });
   }
 
+  /**
+   *  Challenge given payment method
+   *  @param  {Object} paymentMethod The payment method to edit
+   *  @param  {Object} challenge     The challenge value
+   *  @return {Promise}
+   */
+  challengePaymentMethod(paymentMethod, challenge) {
```
Comment (antleblanc) [inline src/payment-method.service.js]: Here is a destructuring assignment suggestion:

```js
  challengePaymentMethod(paymentMethod, challenge) {
    const { original, paymentMethodId } = paymentMethod;
    // if original attribute is present, it means that it's an legacy payment method
    if (original) {
      return this.ovhPaymentMethodLegacy
        .challengePaymentMethod(original, challenge);
    }

     return this.OvhApiMe.Payment().Method().v6().challenge({
      paymentMethodId,
    }, { challenge }).$promise;
  }
```

File: src/payment-method-legacy.js (hunk 1 above)
Comment (FredericEspiau) [reply src/payment-method-legacy.js]:
````js
challengePaymentMean({ id, paymentType }, challenge) {
    return this
      .getPaymentMeanResource(paymentType)
      .challenge({ id }, { challenge }).$promise;
  }
````
or is there something I missed ?

Comment (antleblanc) [reply src/payment-method-legacy.js]: Both are correct ;)

**Comment by antleblanc** [comment]
:tada: `ovh-api-services` has been released/published (https://github.com/ovh-ux/ovh-api-services/releases/tag/v6.26.0)

File: src/payment-method-legacy.js (hunk 1 above)
Comment (antleblanc) [reply src/payment-method-legacy.js]: Could be addressed separately! Merge request;)

**Pull request merged**